File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
var lengthOfLongestSubstring = function(s) {
    const set = new Set();
    let i = 0, maxLen = 0;

    for (let j = 0; j < s.length; j++) {
        while (set.has(s[j])) {
            set.delete(s[i]);
            i++;
        }
        set.add(s[j]);
        maxLen = Math.max(maxLen, j - i + 1);
    }

    return maxLen;
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> set = new HashSet<>();
        int i =0 , maxlen =0;
        for(int j =0; j<s.length(); j++){
            while(set.contains(s.charAt(j))){
                set.remove(s.charAt(i));
                i++;
            }
            set.add(s.charAt(j));
            maxlen = Math.max(maxlen , j -i +1);
        }
        return maxlen;
    }
}